### DISCUSSION TITLE
[BUG] Set-PnPList -BreakRoleInheritance returns "The list is not DocumentLibrary, VersionPolicies is not supported."

### DISCUSSION THREAD

**Discussion by cprovpo** [pnp/powershell#3652] [General]
### Reporting an Issue or Missing Feature
When running Set-PnPList -BreakRoleInheritance -ResetRoleInheritance, the command fails and returns this exception message "The list is not DocumentLibrary, VersionPolicies is not supported."  A user with a very similar issue reported this today on [Stack Overflow here ](https://stackoverflow.com/questions/77601317/set-pnplistpermission-failing-from-azure-automation-runbook-on-list-document-li/77602901#77602901)

### Expected behavior 
The command should result in the list having its inheritance permissions broken from the parent site AND having all existing permissions removed.

### Actual behavior 
When running the script either from VS Code or pwsh.exe, the exception is thrown.  However, if I stop the debugger and run the script manually (by running the selection), it runs successfully.
![image](https://github.com/pnp/powershell/assets/139259660/135b654b-eb31-4aac-b278-17b075533c97)

### Steps to reproduce behavior
$list = "Test List"
Set-PnPList -Identity $list -BreakRoleInheritance -ResetRoleInheritance

Code fails during script runs, but will complete if run as an individual selection.

### What is the version of the Cmdlet module you are running?
PnP.PowerShell 2.2.130 nightly build

### Which operating system/environment are you running PnP PowerShell on?
- [x ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by TyrrellStephen** [comment]
Have you found a solution or workaround for this?

I am getting the same error message with a different cmdlt 'Get-PnPListItem' with PnP 2.4.0. It also works fine when executed interactively at ps prompt (terminal), but not when running as part of a script.

**Reply by renewtx** [reply]
Hi -- did you ever find a resolution for this?  I am also getting the error with Get-PnPListItem ... 

 895 |  … listItems = Get-PnPListItem -List $siteMasterList -Query $query -Conn …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The list is not DocumentLibrary, VersionPolicies is not supported.

Running PowerShell v7.4.5  and PnP.PowerShell  v2.99.35-nightly

**Comment by MatthiasAbt-SVA** [comment]
I have the same problem with different commands (Set-PnPList). For me it looks like a bug somewhere else.

**Comment by MrRCG5** [comment]
I have the same problem.

When adding a content type to a newly created "Generic List" the following script does not work when run as part of an entire script, but the line "Add-PnPContentTypeToList" will work when run individually. 

However, creating a "Document Library" works when run as an entire script.

```
$libraryType = "GenericList"
$libraryUrl = "Lists/AssumptionLog"
$libraryName = "Assumption Log"
$CTID = "0x01001113FF6151313044864C32FD931FF8A50B01"

New-PnPList -Title $libraryName -Template:$libraryType -Url $libraryUrl -OnQuickLaunch:$true -EnableContentTypes:$true -Connection $siteConnection
Add-PnPContentTypeToList -List $libraryName -ContentType $CTID -DefaultContentType -Connection $siteConnection -ErrorAction Stop
**The list is not DocumentLibrary, VersionPolicies is not supported.**
```

**Comment by shaunpitt** [comment]
Hi there - I'd like to add that I am having the same error - but different command.
I'm using PnP.PowerShell 2.99.63

I am running a 500 line script that does a number of things but ultimately to provision a pnp provisioning template.
The templates role is to add some content types to the default doc lib of all sharepoint sites.
After the provisioning has completed depending on the site type - 2 columns will be set to default values (one column is a choice field while the other is a managed metadata field)

Strangely when the code runs as part of the complete script it fails with

`An error occurred: The list is not DocumentLibrary, VersionPolicies is not supported. `

BUT when I run the lines of code in question in isolation it works successfully.

This is the code:
```
# Set default value for a choice field in the document library
  $Context = Get-PnPContext -Connection $siteConn
  $field = Get-PnPField -List $defaultDocumentLibrary -Identity $columnPersonalInformation -Connection $siteConn
  $field.DefaultValue = $defaultPersonalInformation
  $field.Update()
  $Context.ExecuteQuery()
  
  Write-Host "Applying default value for Type of Information................................"
  $fullstringInfoType = "$termGroupName|$termSetName|$businessFunctionValue|$defaultInfoType"
  Set-PnPDefaultColumnValues -List $defaultDocumentLibrary -Field $columnInformationType -Value $fullstringInfoType -Connection $siteConn
  $getInfoTypeDefault = Get-PnPDefaultColumnValues -List $defaultDocumentLibrary -Connection $siteConn | Where-Object { $_.Field -eq $columnInformationType } | Select-Object -ExpandProperty Value
```

I can confirm that the code fails at the Get-PnPField line.

I don't really understand the error - the default document library is a DocumentLibrary, and I am doing nothing with versioning. Am I missing something?

Any insight would be much appreciated
